### PR TITLE
MixinICommandSender: Don't call asICommandSource() when no one implements it.

### PR DESCRIPTION
**Pull request by JBYoshi**
Fixes SpongePowered/SpongeForge#1333.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/core/command/MixinICommandSender.java [hunk 1]
```diff
@@ -30,14 +30,19 @@
...
 
 @Mixin(value = {EntityPlayerMP.class, MinecraftServer.class, RConConsoleSource.class, CommandBlockBaseLogic.class}, targets = IMixinCommandSender.SIGN_CLICK_SENDER)
-public abstract class MixinICommandSender implements ICommandSender, IMixinCommandSender {
+public abstract class MixinICommandSender implements ICommandSender {
 
-    @Override
-    public boolean canUseCommand(int permissionLevel, String commandName) {
-        return CommandPermissions.testPermission(this.asCommandSource(), commandName);
+    @Inject(method = "canUseCommand", at = @At("HEAD"), cancellable = true)
+    public void onCanUseCommand(int permissionLevel, String commandName, CallbackInfoReturnable<Boolean> cir) {
```
Comment (stephan-gh) [inline src/main/java/org/spongepowered/common/mixin/core/command/MixinICommandSender.java]: Injection can be private

**Review by stephan-gh — CHANGES_REQUESTED** [review]

File: src/main/java/org/spongepowered/common/mixin/core/command/MixinICommandSender.java (hunk 1 above)
Comment (JBYoshi) [reply src/main/java/org/spongepowered/common/mixin/core/command/MixinICommandSender.java]: @Minecrell Fixed.

**Comment by JBYoshi** [comment]
Not sure why I closed that before it was merged. I will reopen/remake this later today; ping me if I forget.

**Comment by JBYoshi** [comment]
Reopened. I will need to rebase and fix the build, though.

**EDIT:** Done.

**Comment by JBYoshi** [comment]
Merged in https://github.com/SpongePowered/SpongeCommon/commit/2342f2f7411210dbd4fa6489b8ce71544277adf6.

**Pull request closed without merging**